src/commands/daemon.rs

```diff
@@ -0,0 +1,215 @@
+                .expect("Failed to update status according events");
+        },
+        async {
+            perform_operations_fut
+                .await
+                .expect("Failed to perform operations");
+        },
+        async {
+            server.await.expect("Failed to serve a service");
+        }
+    );
+    Ok(())
+}
+
+fn update_status_according_to_events(client: Client, status: Arc<RwLock<Status>>) {
+    let event_filter = FilterBox::Pipeline(PipelineEventFilter::new());
+    for event in client
+        .listen_for_events(event_filter)
+        .expect("Failed to listen for events")
+    {
+        debug!(event = ?event, "got an event");
+        if let Ok(Event::Pipeline(event)) = event {
+            match event.status {
+                PipelineStatus::Validating => {}
+                PipelineStatus::Rejected(_) => {
+                    status
+                        .write()
+                        .expect("Failed to lock to write rejection timestamp to status")
+                        .tx_is_rejected();
+                }
+                PipelineStatus::Committed => {
+                    status
+                        .write()
+                        .expect("Failed to lock to write commit timestamp to status")
+                        .tx_is_committed();
+                }
+            }
+        } else {
+            warn!("TX with unknown status");
+            status
+                .write()
+                .expect("Failed to lock to write unknown status")
+                .tx_is_unknown()
+        }
+    }
+}
+
+fn perform_operations(
+    client: Client,
+    status: Arc<RwLock<Status>>,
+    interval: Duration,
+    mut operations: HashMap<Operation, usize>,
+) {
+    let alice_id = AccountId::from_str("alice@wonderland").expect("Failed to make Alice id");
+    let wonderland_id =
+        DomainId::new(Name::from_str("wonderland").expect("Failed to create Wodnerland name"));
+    while !operations.is_empty() {
+        operations.retain(|op, count| {
+            let start_time = Instant::now();
+            debug!(operation = ?op, count = ?count, "perform operation");
+            let instructions =
+                make_instruction_by_operation(op, alice_id.clone(), wonderland_id.clone(), *count);
+            let res = client.submit_all(instructions);
+            let elapsed = Instant::now().duration_since(start_time);
+            if elapsed < interval {
+                thread::sleep(interval - elapsed);
+            }
+            if let Err(err) = res {
+                warn!("Submit failed: {}", err);
+                true
+            } else {
+                status
+                    .write()
+                    .expect("Failed to lock to write status")
+                    .tx_is_sent();
+                *count -= 1;
+                *count != 0
+            }
+        });
+    }
+}
+
+fn submit_empty_transactions(client: Client, status: Arc<RwLock<Status>>, interval: Duration) {
```

Comment:
```suggestion
fn submit_empty_transaction(client: Client, status: Arc<RwLock<Status>>, interval: Duration) {
```

Effectively it submits the only one transaction.

Reply:
No, wait, it actually submits transactions in a loop.